pyfr/plugins/acoustics.py

```diff
@@ -0,0 +1,2106 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict, namedtuple
+from mpi4py import MPI
+import numpy as np
+import os
+import re
+import sys
+
+from pyfr.inifile import Inifile
+from pyfr.mpiutil import get_comm_rank_root, get_mpi
+from pyfr.nputil import CubicSplineFit, fuzzysort, LinearFit, npeval
+from pyfr.plugins.base import BasePlugin
+from pyfr.writers.native import NativeWriter
+
+
+def open_csv(fname, header=None, mode='w'):
+    # Open for mode
+    outf = open(fname, mode)
+    # Output a header if required
+    if os.path.getsize(fname) == 0 and header:
+        print(header, file=outf)
+    return outf
+
+def write_fftdata(fname, *indata, header='', mode='a'):
+    #open the file
+    outf = open_csv(fname, ''.join(header), mode)
+    #prepare the data array
+    data = np.array(indata)
+    dshape = data.shape
+    #do write
+    for i in range(dshape[1]):
+        print(*data[:, i].ravel(), sep=' ', file=outf)
+    outf.flush()
+    return
+
+def realfft(udata_):
```

Comment:
Function is mostly pointless and can likely be inlined.

Reply:
used several times and this is better wrapper to suit our needs. In the future you can be performing more rfft's in another acoustic solver such as time domain or so. Less code writing.